datasets/inception_subset.py

```diff
@@ -0,0 +1,174 @@
+########################################################################################################
+#                                                                                                      #
+#                           Inception Subset                                                           #
+#                                                                                                      #
+########################################################################################################
+
+# TORCH
+import torch
+from torch.utils.data import Dataset, TensorDataset
+
+# Python Packages
+import numpy as np
+# import glob
+from datetime import datetime
+from typing import List, Type, Tuple, Iterator
+from collections import defaultdict  # Nested dictionaries in python
+
+# SKLearn
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import StratifiedKFold
+
+# PNN SPECIFIC
+from datasets.labels import labels
+from train import parameters
+
+
+class Modalities(Dataset):
+    """
+    A dataset class that lets the user decides which modalities to use.
+    """
+
+    def __init__(self, root_dir: str,
+                 exp_name: str,
+                 start_date,
+                 end_date,
+                 **plant_files_dict):
+        """
+        :param root_dir: path to the experiment directory
+        :param exp_name: the experiment we want to use
+        :param mods: modalities to be in the dataset, as dictionaries of initialization arguments
+
+        """
+        self.experiment = exp_name
+        self.num_plants = len(labels[exp_name])
+        self.modalities = defaultdict(dict)
+        self.plant_files_dict = plant_files_dict
+        self.num_plants = len(labels[self.experiment])
+        # todo: fix used plants to be more generic
+        self.used_plants = [plant for plant in range(self.num_plants)]
```

Comment:
- [ ] use the `list` operator
```suggestion
        self.used_plants: List[int] = list(range(self.num_plants))
```